### PR TITLE
Modifications needed for HF sync 

### PR DESCRIPTION
**Pull request by EiffL**
This PR implements a number of consistency fixes to enable the automated upload to HuggingFace. 

These modifications have been tested, and the HF previews are being updated.

#### No renaming/reformatting of native fields

Apart from ra, dec, object_id, image, spectrum, lightcurve, **we do not rename variables from the original surveys**, not even to normalize redshift between Z, z, 'redshift'. 

This is important because **we do not provide documentation for what these fields mean**, and instead direct the user to the original survey documentation. There is no such thing as a "redshift" for instance, any redshift measurement is the output of some algorithm, same for magnitude and such. It is important that the user go back to the survey-specific definition, otherwise we need to do a lot more work to add these descriptions ourselves.

#### Reversionning all datasets to 1.0.0

When  we make updates after this, we can incrementally increase this version number

### CODE REVIEW

File: scripts/btsbot/btsbot.py [hunk 1]
```diff
@@ -268,10 +265,10 @@ def _generate_examples(self, files, object_ids=None):
                     example = {
                         'image': [
                             {
-                                'band': data['band'],
+                                'band': data['band'][i],
                                 'view': view,
                                 'array': data['image_triplet'][i, :, :, j],
-                                'scale': data['image_scale'],
+                                'scale': data['image_scale'][i],
```
Comment (mb010) [inline scripts/btsbot/btsbot.py:271]: Has this been tested?

File: scripts/btsbot/btsbot.py [hunk 2]
```diff
@@ -68,13 +68,11 @@
 
 _LICENSE = "CC BY 4.0"
 
-_VERSION = "10.0.0"
+_VERSION = "1.0.0"
```
Comment (mb010) [inline scripts/btsbot/btsbot.py:71]: Note: It was v10 to match the official release of BTSbot data.

File: scripts/cfa/cfa.py [hunk 3]
```diff
@@ -18,14 +18,14 @@
...
-_VERSION = "0.0.1"
+_VERSION = "1.0.0"
 
 _HOMEPAGE = "https://lweb.cfa.harvard.edu/supernova/"
 _LICENSE = "CC BY 4.0"
 
 # Features common across all CFA datasets
 _STR_FEATURES = ["object_id", "obj_type"]
-_FLOAT_FEATURES = ["ra", "dec"]
+_FLOAT_FEATURES = []
```
Comment (mb010) [inline scripts/cfa/cfa.py:28]: Are these not used?

File: scripts/gz10/gz10.py [hunk 4]
```diff
@@ -93,9 +93,7 @@ def _info(self):
         features = datasets.Features(
             {
                 "gz10_label": datasets.Value("int32"),
-                "ra": datasets.Value("float32"),
-                "dec": datasets.Value("float32"),
```
Comment (mb010) [inline scripts/gz10/gz10.py:97]: Why are we dropping these keys? I think they should be included for cross matching.

File: scripts/gz10/gz10.py [hunk 5]
```diff
@@ -150,9 +148,7 @@ def _generate_examples(self, files, object_ids=None):
 
                     example = {
                         "gz10_label": data["ans"][i].astype(np.int32),
-                        "dec": data["dec"][i].astype(np.float32),
-                        "ra": data["ra"][i].astype(np.float32),
```
Comment (mb010) [inline scripts/gz10/gz10.py:154]: Why are we dropping these keys? I think they should be included for cross matching.

File: scripts/manga/manga.py [hunk 6]
```diff
@@ -81,9 +81,6 @@ def _info(cls):
 
         # add metadata features
         features['object_id'] = Value("string")
-        features['ra'] = Value("float32")
-        features['dec'] = Value("float32")
```
Comment (mb010) [inline scripts/manga/manga.py:85]: These shouldn't be dropped unless I'm missing something.

File: scripts/snls/snls.py [hunk 7]
```diff
@@ -61,16 +61,14 @@
...
+_VERSION = "1.0.0"
 
 _STR_FEATURES = [
     "object_id",
     "obj_type"
 ]
 
 _FLOAT_FEATURES = [
-    "ra", 
-    "dec",
```
Comment (mb010) [inline scripts/snls/snls.py:73]: These shouldn't be dropped unless I'm missing something.

File: scripts/ps1_sne_ia/ps1_sne_ia.py [hunk 8]
```diff
@@ -88,7 +86,7 @@ class PS1SNIa(datasets.GeneratorBasedBuilder):
         datasets.BuilderConfig(
             name="ps1_sne_ia",
             version=VERSION,
-        data_files=DataFilesPatternsDict.from_patterns({"train": ["./healpix=*/*.hdf5"]}), # This seems fairly inflexible. Probably a massive failure point.
+        data_files=DataFilesPatternsDict.from_patterns({"train": ["ps1_sne_ia/healpix=*/*.hdf5"]}), # This seems fairly inflexible. Probably a massive failure point.
```
Comment (mb010) [inline scripts/ps1_sne_ia/ps1_sne_ia.py:89]:
```suggestion
        data_files=DataFilesPatternsDict.from_patterns({"train": ["ps1_sne_ia/healpix=*/*.hdf5"]}),
```
Removing comment.

File: scripts/ssl_legacysurvey/ssl_legacysurvey.py [hunk 9]
```diff
@@ -92,7 +92,7 @@ class SSLLegacySurvey(datasets.GeneratorBasedBuilder):
             name="stein_et_al",
             version=VERSION,
             data_files=DataFilesPatternsDict.from_patterns(
-                {"train": ["*/healpix=*/*.hdf5"]}
+                {"train": ["north/healpix=*/*.hdf5"]}
```
Comment (mb010) [inline scripts/ssl_legacysurvey/ssl_legacysurvey.py:95]: not `ssl_legacysurvey`?

File: scripts/swift_sne_ia/swift_sne_ia.py [hunk 10]
```diff
@@ -60,16 +60,14 @@
...
+_VERSION = "1.0.0"
 
 _STR_FEATURES = [
     "object_id",
     "obj_type"
 ]
 
 _FLOAT_FEATURES = [
-    "ra", 
-    "dec",
```
Comment (mb010) [inline scripts/swift_sne_ia/swift_sne_ia.py:72]: These shouldn't be dropped unless I'm missing something.

**Review by mb010 — CHANGES_REQUESTED** [review]
See comments.

There is a repeating question: Why is this PR trying to drop `ra` and `dec` in many of the datasets?

File: scripts/gz10/gz10.py (hunk 5 above)
Comment (EiffL) [reply scripts/gz10/gz10.py:154]: no, cross matching is not done at that level

File: scripts/snls/snls.py (hunk 7 above)
Comment (EiffL) [reply scripts/snls/snls.py:73]: we are not including ra, dec as features

File: scripts/ssl_legacysurvey/ssl_legacysurvey.py (hunk 9 above)
Comment (EiffL) [reply scripts/ssl_legacysurvey/ssl_legacysurvey.py:95]: no, it's called north

File: scripts/btsbot/btsbot.py (hunk 2 above)
Comment (EiffL) [reply scripts/btsbot/btsbot.py:71]: yeah, but this refers to the MMU version of the dataset

File: scripts/btsbot/btsbot.py (hunk 1 above)
Comment (EiffL) [reply scripts/btsbot/btsbot.py:271]: yes

File: scripts/ssl_legacysurvey/ssl_legacysurvey.py (hunk 9 above)
Comment (EiffL) [reply scripts/ssl_legacysurvey/ssl_legacysurvey.py:95]: because in principle we could have also added a 'south'

**Review by mb010 — APPROVED** [review]
LGTM

**Pull request merged**